activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -638,8 +638,27 @@ def find_last(limit)
       end
 
       def ordered_relation
-        if order_values.empty? && !_order_columns.empty?
-          order(_order_columns.map { |column| table[column].asc })
+        if order_values.empty?
+          if !_order_columns.empty?
+            return order(_order_columns.map { |column| table[column].asc })
+          end
+
+          if ActiveRecord.raise_on_missing_required_finder_order_columns
+            raise MissingRequiredOrderError, <<~MSG.squish
+              Relation has no order values, and #{model} has no order columns to use as a default.
```

Comment:
```suggestion
              Relation has no order values, and #{model.name} has no order columns to use as a default.
```

`model` is a AR class here right? I think we should just use the name, because otherwise `AR::Base.inspect` can be super long and hard to read.